server/src/main/java/com/google/fhir/proxy/BearerAuthorizationInterceptor.java

```diff
@@ -134,19 +135,21 @@ private RSAPublicKey fetchAndDecodePublicKey() throws IOException {
       return (RSAPublicKey) keyFactory.generatePublic(keySpec);
     } catch (URISyntaxException e) {
       ExceptionUtil.throwRuntimeExceptionAndLog(
-          logger, "Error in token issuer URI " + tokenIssuer, e);
+          logger, "Error in token issuer URI " + tokenIssuer, e, AuthenticationException.class);
     } catch (NoSuchAlgorithmException e) {
-      ExceptionUtil.throwRuntimeExceptionAndLog(logger, "Invalid algorithm " + keyAlgorithm, e);
+      ExceptionUtil.throwRuntimeExceptionAndLog(
+          logger, "Invalid algorithm " + keyAlgorithm, e, AuthenticationException.class);
     } catch (InvalidKeySpecException e) {
-      ExceptionUtil.throwRuntimeExceptionAndLog(logger, e);
+      ExceptionUtil.throwRuntimeExceptionAndLog(
```

Comment:
add "Invalid key spec here" or something similar to match error messages above

Reply:
Done.